File: 2245-destroying-asteroids/destroying-asteroids.cs
public class Solution {
    public bool AsteroidsDestroyed(int mass, int[] asteroids) {
        Array.Sort(asteroids);

        long m = mass;
        foreach (int x in asteroids) {
            if (m < x) {
                return false;
            }
            m += x;
        }

        return true;
    }
}

File: 2245-destroying-asteroids/destroying-asteroids.go
func asteroidsDestroyed(mass int, asteroids []int) bool {
	sort.Ints(asteroids)

	m := int64(mass)

	for _, x := range asteroids {
		if m < int64(x) {
			return false
		}
		m += int64(x)
	}
	return true
}